numpyro/primitives.py

```diff
@@ -80,7 +80,21 @@ def __call__(self, *args, **kwargs):
             return self.fn(*args, **kwargs)
 
 
-def sample(name, fn, obs=None, rng_key=None, sample_shape=(), infer=None):
+def _masked_observe(name, fn, obs, obs_mask, **kwargs):
+    # Split into two auxiliary sample sites.
+    with numpyro.handlers.mask(mask=obs_mask):
+        observed = sample(f"{name}_observed", fn, **kwargs, obs=obs)
+    with numpyro.handlers.mask(mask=(not obs_mask if isinstance(obs_mask, bool) else ~obs_mask)):
```

Comment:
I had this issue too, and ended up using `___ ^ False` for type safe negation

Reply:
Nice! Thank you!